### PR TITLE
[ts-sdk-v2] Add getTransactions to the sdk-v2 transaction API

### PR DESCRIPTION
**Pull request by heliuchuan**
### Description
Adds getTransactions to the sdk API

### Test Plan
None - blocked on transaction submission

### CODE REVIEW

File: ecosystem/typescript/sdk_v2/src/internal/transaction.ts [hunk 1]
```diff
@@ -7,8 +7,24 @@
...
+import { paginateWithCursor } from "../utils/paginate_with_cursor";
+
+export async function getTransactions(args: {
+  aptosConfig: AptosConfig;
+  options?: PaginationArgs;
+}): Promise<TransactionResponse[]> {
+  const { aptosConfig, options } = args;
+  const data = await paginateWithCursor<{}, TransactionResponse[]>({
+    url: aptosConfig.getRequestUrl(AptosApiType.FULLNODE),
+    endpoint: `transactions`,
```
Comment (0xmaayan) [inline ecosystem/typescript/sdk_v2/src/internal/transaction.ts:21]: nit
```suggestion
    endpoint: "transactions",
```

File: ecosystem/typescript/sdk_v2/src/internal/transaction.ts [hunk 2]
```diff
@@ -7,8 +7,24 @@
...
+    originMethod: "getTransactions",
+    params: { start: options?.start, limit: options?.limit },
+    overrides: { ...aptosConfig.clientConfig },
```
Comment (0xmaayan) [inline ecosystem/typescript/sdk_v2/src/internal/transaction.ts:24]: once https://github.com/aptos-labs/aptos-core/pull/10216/files#diff-5a5120925d00a2725ac86e969e51b41bb7cd9276bd875eb54d655d5a5f2aaae0R75, the data passed into the `paginateWithCursor()` changes

**Review by 0xmaayan — APPROVED** [review]
LGTM. need to keep an eye on changes for the data we are passing into the `paginateWithCursor()` function

**Review by gregnazario — APPROVED** [review]

File: ecosystem/typescript/sdk_v2/src/internal/transaction.ts (hunk 2 above)
Comment (heliuchuan) [reply ecosystem/typescript/sdk_v2/src/internal/transaction.ts:24]: right, right, i'll wait for that to go in before merging

**Pull request merged**